repose-aggregator/components/filters/uri-normalization-filter/src/test/scala/org/openrepose/filters/urinormalization/normalizer/MediaTypeNormalizerTest.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.urinormalization.normalizer
+
+import org.junit.runner.RunWith
+import org.openrepose.commons.utils.http.CommonHttpHeader
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.filters.urinormalization.config.MediaType
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+import org.springframework.mock.web.MockHttpServletRequest
+
+@RunWith(classOf[JUnitRunner])
+class MediaTypeNormalizerTest extends FunSpec with BeforeAndAfter with Matchers {
+
+  val configuredMediaTypes = Seq(
+    new MediaType().withName("application/xml").withVariantExtension("xml").withPreferred(true)
+  )
+
+  var mediaTypeNormalizer: MediaTypeNormalizer = _
+
+  implicit class WithableMediaType(val mt: MediaType) extends MediaType {
```

Comment:
The more I looked at it, the less I liked it in the end. I had some other ideas, but I've lost them now. I'll have to try to come up with something more general at some point.

At least I got to play with implicits here =)